js/noms/src/value-store.js

```diff
@@ -143,15 +151,15 @@ class PendingChunk {
 interface Cache<T> {  // eslint-disable-line no-undef
   entry(hash: Hash): ?CacheEntry<T>;  // eslint-disable-line no-undef
   get(hash: Hash): ?T;  // eslint-disable-line no-undef
-  add(hash: Hash, size: number, value: T): void;  // eslint-disable-line no-undef
+  add(hash: Hash, size: Promise<number>, value: T): void;  // eslint-disable-line no-undef
```

Comment:
I'm observing that `Cache` is only ever used with `T` as a `Promise<?Value>`

and making size a Promise makes it look weird that value isn't also a Promise.

It's probably not worth going down this route, but you could delete the generics on this and CacheEntry and make it `add(Hash, Promise<number>, Promise<Value>)` etc.